R/client-connect.R

```diff
@@ -83,42 +92,29 @@ connectClient <- function(service, authInfo) {
     },
 
     deployApplication = function(application, bundleId = NULL) {
-      path <- paste("/applications/", application$id, "/deploy", sep = "")
-      json <- list()
-      json$bundle <- as.numeric(bundleId)
-      POST_JSON(service, authInfo, path, json)
-    },
-
-    configureApplication = function(applicationId) {
-      GET(
+      path <- v1_url("content", application$guid, "deploy")
+      POST_JSON(
         service,
         authInfo,
-        paste(
-          "/applications/",
-          applicationId,
-          "/config",
-          sep = ""
-        )
+        path,
+        json = list(bundle_id = bundleId)
       )
     },
 
     getApplication = function(applicationId, deploymentRecordVersion) {
-      GET(service, authInfo, paste0("/applications/", applicationId))
+      GET(service, authInfo, unversioned_url("applications", applicationId))
     },
 
     waitForTask = function(taskId, quiet = FALSE) {
-      first <- 0
-      wait <- 1
+      path <- v1_url("tasks", taskId)
+      query <- list(first = 0, wait = 1)
+
       while (TRUE) {
-        path <- paste0(
-          "/v1/tasks/",
-          taskId,
-          "?first=",
-          first,
-          "&wait=",
-          wait
-        )
-        response <- GET(service, authInfo, path)
+        # ick, manual url construction
```

Comment:
I started refactoring this because I thought we could just let `httr::GET()` handle the query string assembly, but as it turns out, `GET()` here is not from `httr`.

Reply:
This might be a separate PR regardless, but IIUC one of the reasons we rolled our own `GET` was to support configurations that are no longer a real issue (IIRC, though my searching just now didn't turn up anything elucidating, there was a period where we needed to support less-than-standardly-secure http connections). I did find https://github.com/rstudio/rsconnect/pull/762 which started reducing the backends...)